cranelift-codegen/meta/src/cdsl/operands.rs

```diff
@@ -19,18 +19,34 @@ use crate::cdsl::typevar::TypeVar;
 #[derive(Clone, Debug)]
 pub(crate) struct Operand {
     pub name: &'static str,
-    doc: Option<String>,
+    doc: Option<&'static str>,
     pub kind: OperandKind,
 }
 
 impl Operand {
+    pub fn new(name: &'static str, kind: impl Into<OperandKind>) -> Self {
+        Self {
+            name,
+            doc: None,
+            kind: kind.into(),
+        }
+    }
+
+    pub fn with_doc(name: &'static str, kind: impl Into<OperandKind>, doc: &'static str) -> Self {
+        Self {
+            name,
+            doc: Some(doc),
+            kind: kind.into(),
+        }
+    }
```

Comment:
I would have written this in a different API style:
```
pub fn with_doc(self, doc: &'static str) -> Self {
  Self {
    doc: Some(doc),
    ..self
  }
}
```
So that callers do `Operand::new(...).with_doc(...)`. Just a style thing so feel free to ignore, but it avoids the duplicated constructor code.